### PR TITLE
fix: type for plain action creator. Fixes #143

### LINKED ISSUE #143: createAction with only a type parameter results in function, which allows any arguments
```typescript
const myAction = createAction('MY_ACTION');
myAction('woot', 14, null, undefined); // Does not complain
```
This is doing the same thing as far as I can see, but it complains correctly:
```typescript
export const myAction = createAction('MY_ACTION', (resolve) => () => resolve());
myAction('woot', 14, null, undefined); // Error
```

### PR DESCRIPTION
**Pull request by SantoJambit**
I had to create a new type `ExactActionCreator` since the `ActionCreator` type did not specify TCallable. Modifying `ActionCreator` didn't seem viable since it would introduce a breaking change.

Not sure about the name of this new type. maybe you have a better idea.

### CODE REVIEW

**Comment by SantoJambit** [comment]
I'm assuming the deploy fails are not related to my PR.

**Comment by thebrodmann** [comment]
@SantoJambit Thank you for submitting this PR. Can you add a test case to DTS tests for this PR? It can check for type checker failure on application of arguments to a nullary function.

**Comment by SantoJambit** [comment]
Correct me if I'm wrong, but I thought the snapshot already takes care of that:

```diff
-exports[`createActionCreator createActionCreator('[Todo] truncate') (type) should match snapshot 1`] = `"(<_T>(...args: any[]) => { type: \\"[Todo] truncate\\"; }) & { type: \\"[Todo] truncate\\"; toString(): \\"[Todo] truncate\\"; }"`;
+exports[`createActionCreator createActionCreator('[Todo] truncate') (type) should match snapshot 1`] = `"ExactActionCreator<\\"[Todo] truncate\\", () => { type: \\"[Todo] truncate\\"; }>"`;
```

This shows, that the returned function may not receive any args.

**Comment by SantoJambit** [comment]
I think I get what you want.. you want typescript errors in the dts.spec file. As long as this builds, I can try something.

**Comment by SantoJambit** [comment]
@thebrodmann hope this is what you wanted.

File: src/__tests__/__snapshots__/create-action-creator.dts.spec.ts.snap [hunk 1]
```diff
@@ -1,35 +1,120 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
-exports[`createActionCreator createActionCreator(
+exports[`createActionCreator const todoAdd = createActionCreator(
       '[Todo] add',
       // tslint:disable-next-line:no-inferrable-types
       resolve => (name: string, completed: boolean = false) =>
         resolve({ name, completed })
-    ) (type) should match snapshot 1`] = `"((name: string, completed?: boolean) => { type: \\"[Todo] add\\"; payload: { name: string; completed: boolean; }; }) & { type: \\"[Todo] add\\"; toString(): \\"[Todo] add\\"; }"`;
+    ) (type) should match snapshot 1`] = `"any"`;
```
Comment (thebrodmann) [inline src/__tests__/__snapshots__/create-action-creator.dts.spec.ts.snap]: This is not true. I think you should snapshot like follow above and similar cases:

```ts
const a = createActionCreator('A')

// @dts-jest:pass:snap
a
```
```

Comment (thebrodmann) [reply src/__tests__/__snapshots__/create-action-creator.dts.spec.ts.snap]: so let's create each action creator twice. One for snapshot and another for testing created action creators.

Comment (SantoJambit) [reply src/__tests__/__snapshots__/create-action-creator.dts.spec.ts.snap]: as you wish.

**Comment by SantoJambit** [comment]
@thebrodmann anything missing?

**Comment by thebrodmann** [comment]
Absolutely not. Sorry for the latency. Let's go for the release of this honor.

**Comment by SantoJambit** [comment]
Thanks!

**Pull request merged**